File: golang/fizzbuzz.go
package fizzbuzz

import (
	"strconv"
)

type FizzBuzz struct {
}

func NewFizzBuzz() FizzBuzz {
	fb := FizzBuzz{}
	return fb
}

func (fb FizzBuzz) Do(i int) string {
	if i%3 == 0 {
		if i%5 == 0 {
			return "FizzBuzz"
		}
		return "Fizz"
	} else if i%5 == 0 {
		return "Buzz"
	}
	return strconv.Itoa(i)
}

File: ruby/lib/fizzbuzz.rb
class FizzBuzz
  def fizzbuzz(i)
    if i % 3 == 0
      return "FizzBuzz" if i % 5 == 0
      return "Fizz"
    elsif i % 5 == 0
      return "Buzz"
    end
    return i.to_s
  end
end
